From 3e22d345fac2c3e4d35590e2bc9e500ce0564aea Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Thu, 26 Jul 2018 10:50:51 +0800
Subject: [PATCH 1344/1345] driver: perf: mvebu-spmu: initialize task_ctx_nr
 field of struct pmu

Because the upstream commit 8dc85d547285("perf: Multiple task
contexts"), it is need to initialize task_ctx_nr field of struct
pmu with perf_invalid_context. Otherwise, there will be a warning
trace when register pmu instance with function perf_pmu_register().

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/perf/mvebu-spmu.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/perf/mvebu-spmu.c b/drivers/perf/mvebu-spmu.c
index ee5b0c9..b01ab12 100644
--- a/drivers/perf/mvebu-spmu.c
+++ b/drivers/perf/mvebu-spmu.c
@@ -560,6 +560,7 @@ static int mvebu_spmu_pmu_init(struct mvebu_spmu *spmu)
 	spmu->pmu.start = mvebu_spmu_start;
 	spmu->pmu.stop = mvebu_spmu_stop;
 	spmu->pmu.read = mvebu_spmu_read;
+	spmu->pmu.task_ctx_nr = perf_invalid_context;
 
 	err = perf_pmu_register(&spmu->pmu, name, -1);
 	if (err)
-- 
1.7.9.5

